### DISCUSSION TITLE
请问对于类SQL指令要求的理解这样的么？

### DISCUSSION THREAD

**Discussion by huwenjing0819** [liusy58/NJU_AdvancedProgramming_SP22#11] [Q&A] [answered]
比如，在管理员菜单界面输入1，我需要生成SELECT * FROM commodity并且写入到command.txt文件中，然后可以在代码中进一步加工这个字符串，比如改成SELECT good FROM commodity并将加工后的字符串传入小函数中，小函数需要能够解析传入的字符串并且实现相应功能？

**Comment by Price1999a** [comment, marked as answer]
>比如，在管理员菜单界面输入1，我需要生成SELECT * FROM commodity并且写入到command.txt文件中

是的

>然后可以在代码中进一步加工这个字符串，比如改成SELECT good FROM commodity并将加工后的字符串传入小函数中，小函数需要能够解析传入的字符串并且实现相应功能？

我感觉这个操作似乎没什么必要？你需要的是尝试解析 `SELECT * FROM commodity`。 当然如果你认为解析`SELECT good FROM commodity`更好，那么也许你可以直接生成`SELECT good FROM commodity`写入`command.txt`。（不考虑格式问题）

**Reply by Price1999a** [reply]
就是 SQL里面 `SELECT good FROM commodity`格式其实是不对的。。
[SQL资料](https://www.runoob.com/sql/sql-select.html)

当然我们这节课不是数据库，所以我们不会在这个地方太纠结

**Reply by huwenjing0819** [reply]
好的，谢谢🙏

**Comment by Randy-13** [comment]
请问一下，所以这个操作的意义是用来记录所有的操作过程吗（就好像一个操作历史记录一样）？
